src/hooks/useExpression.js

```diff
@@ -0,0 +1,89 @@
+import { useState } from 'react';
+
+import { arithmetic } from '../utils/arithmetic';
+
+import { ERROR_MESSAGE, ERROR_TEXT } from '../constants/error';
+
+const MAX_LENGHT = 3;
+
+const useExpression = () => {
+  const [totalText, setTotalText] = useState('0');
+  const [operands, setOperands] = useState([]);
+  const [operations, setOperations] = useState([]);
+
+  const resetCalculator = (result) => {
+    if (!result) {
+      setOperands([]);
+      setOperations([]);
+    }
+
+    setOperands([result]);
+    setOperations([]);
+  };
+
+  const insertErrorMessage = (message) => {
+    if (message) {
+      alert(message);
+    }
+
+    resetCalculator();
+    setTotalText(ERROR_TEXT);
+  };
+
+  const insertOperation = (buttonValue) => {
+    if (!operands[0]) {
+      alert(ERROR_MESSAGE.NUMBER_FIRST);
+      return;
+    }
+
+    setOperations([buttonValue]);
+    setTotalText(totalText + buttonValue);
+  };
+
+  const insertOperand = (buttonValue) => {
+    const counter = operations.length;
+
+    if (operands[counter]?.length >= MAX_LENGHT) {
+      alert(ERROR_MESSAGE.NUMBER_OF_EXCEPTION);
+      return;
+    }
+
+    operands[counter] = operands[counter] ? operands[counter] + buttonValue : buttonValue;
+
+    setOperands([...operands]);
+    setTotalText(totalText === '0' || totalText === ERROR_TEXT ? buttonValue : totalText + buttonValue);
+  };
+
+  const insertResult = (result) => {
+    if (result === Infinity) {
+      insertErrorMessage();
+      return;
+    }
+
+    resetCalculator(result);
+    setTotalText(String(result));
+  };
+
+  const handleExpressionWrite = (event) => {
+    const buttonValue = event.target.value;
+
+    isNaN(buttonValue) ? insertOperation(buttonValue) : insertOperand(buttonValue);
```

Comment:
아래와 같은 내용인데요, 값을 사용하지 않는다면 제어문을 사용하는게 의미전달이 명확합니다.

반환된 값을 개발자가 깜빡하고 사용을 안한건지, 의도된건지 파악이 어렵거든요 😄 

```suggestion
    const buttonValue = event.target.value;

    if (isNaN(buttonValue)) {
      insertOperation(buttonValue)
      return;
    }
    
    insertOperand(buttonValue);
```